reward/src/lib.rs

```diff
@@ -0,0 +1,532 @@
+            );
+            term::blank();
+
+            let decimals = rt.block_on(token.get_decimals()).ok()?.into();
+            Some(decimals)
+        })
+        .ok_or_else(|| anyhow!("Couldn't determine token's decimals"))?;
+
+    let amount = options
+        .amount
+        .or_else(|| {
+            let mut amount = U256::from(0_u64);
+            while amount.is_zero() {
+                let amt: String = term::text_input("Total reward amount?", None).unwrap();
+                amount = amount_to_u256(&amt, decimals).ok()??;
+                if amount.is_zero() {
+                    term::warning("Reward amount cannot be zero");
+                }
+            }
+            Some(amount)
+        })
+        .ok_or_else(|| anyhow!("Couldn't parse amount as u256"))?;
+
+    let strategy = options
+        .strategy
+        .or_else(|| {
+            term::select_with_prompt(
+                "How should the reward be distributed?",
+                &["Equally", "Weighted"],
+                &"Equally",
+            )
+            .unwrap()
+            .parse()
+            .ok()
+        })
+        .ok_or_else(|| anyhow!("Couldn't determine distribution strategy"))?;
+
+    let rewards = calculate_rewards(strategy, amount, &all_ens)?;
+
+    // show summary and aggregate payments
+    term::blank();
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Address"),
+        term::format::bold("Reward"),
+        term::format::bold("ENS"),
+    ]);
+    let mut receivers = Vec::new();
+    let mut amounts = Vec::new();
+    for (ens, reward) in rewards.iter() {
+        let resolver: Result<PublicResolver<SignerMiddleware<_, _>>, _> =
+            rt.block_on(PublicResolver::get(ens, signer.clone()));
+
+        // if resolver doesn't exist, we just skip this one
+        if resolver.is_err() {
+            table.push([
+                term::format::italic(term::format::negative("Missing")),
+                term::format::tertiary(u256_to_amount(*reward, decimals)?),
+                term::format::secondary(ens),
+            ]);
+            continue;
+        }
+        let resolver = resolver.unwrap();
+
+        let address = rt
+            .block_on(resolver.address(ens))?
+            .ok_or_else(|| anyhow!("Couldn't get Address of ENS"))?;
+
+        receivers.push(address);
+        amounts.push(*reward);
+
+        table.push([
+            term::format::highlight(address),
+            term::format::tertiary(u256_to_amount(*reward, decimals)?),
+            term::format::secondary(ens),
+        ]);
+    }
+    table.render();
+    term::blank();
+
+    // exit now if this was a dry run
+    if options.dry {
+        return Ok(());
+    }
+
+    if term::confirm("Do you wish to proceed?") {
+        let sum = amounts
+            .iter()
+            .fold(Some(U256::from(0_u64)), |sum, amt| {
+                sum.map(|s| s.checked_add(*amt)).unwrap()
```

Comment:
We're dealing with user input here, so I'd return an error if the checked_add fails, eg. `amount too large`